src/org/labkey/test/components/domain/AdvancedFieldSetting.java

```diff
@@ -0,0 +1,41 @@
+package org.labkey.test.components.domain;
+
+import org.labkey.test.components.PropertiesEditor;
+
+import java.util.function.BiConsumer;
+
+public class AdvancedFieldSetting<T>
```

Comment:
This approach seems interesting; would definitely like to talk a bit about how it works